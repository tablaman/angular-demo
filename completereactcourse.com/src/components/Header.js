import React from "react";
import { NavLink } from "react-router-dom";

// Header
const Header = () => (
  <React.Fragment>
    <h1>Expensify yey!</h1>
    <NavLink to="/" activeClassName="is-active">Dashboard</NavLink>
    <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
    <NavLink to="/edit" activeClassName="is-active">Edit Expense</NavLink>
    <NavLink to="/help" activeClassName="is-active">Help</NavLink>
  </React.Fragment>
);

export default Header;
