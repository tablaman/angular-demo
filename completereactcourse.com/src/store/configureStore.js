import {
  createStore,
  combineReducers
} from 'redux';
import expensesReducer from '../reducers/expenses';
import filtersReducer from '../reducers/filters';
import helpReducer from '../reducers/help';

export default () => {
  // store creation
  const store = createStore(
    combineReducers({
      expenses: expensesReducer,
      filters: filtersReducer,
      help: helpReducer
    }),
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  );

  return store;
};
