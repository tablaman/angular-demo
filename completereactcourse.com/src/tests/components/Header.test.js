import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const store = configureStore([
  thunk,
])();

test('should render Header corretly', () => {

  const wrapper = shallow(<Header store={store} startLogout={() => {}}/>).dive();

  expect(wrapper).toMatchSnapshot();

  // expect(wrapper.find('h1').length).toBe(1);
  // expect(wrapper.find('h1').text()).toBe("Expensify yey!");
  
})

// test('should call startLogout on button click', () => {
//   const startLogout = jest.fn();
//   const wrapper = shallow(<Header store={store} startLogout={startLogout} />).dive();
//   wrapper.find('button').simulate('click');
//   expect(startLogout).toHaveBeenCalled();
// })
// test('should call startLogout on button click', () => {
//   const startLogin = jest.fn();
//   const wrapper = shallow(<Header store={store} startLogin={startLogin} />).dive();
//   wrapper.find('button').simulate('click');
//   expect(startLogin).toHaveBeenCalled();
// })