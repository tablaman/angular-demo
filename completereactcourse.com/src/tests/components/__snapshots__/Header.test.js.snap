// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header corretly 1`] = `
<React.Fragment>
  <h1>
    Expensify yey!
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Create Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/edit"
  >
    Edit Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
    Help
  </NavLink>
  <button
    onClick={[Function]}
  >
    logout
  </button>
</React.Fragment>
`;
